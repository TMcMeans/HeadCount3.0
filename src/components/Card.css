@import url('https://fonts.googleapis.com/css?family=Poppins');

.red {
  color: grey;
}

.red:before {
  content: '- ';
}

.green {
  color: #4a314d;
  /* font-weight: bold; */
}

.green:before {
  content: '+ ';
}

.card {
  border: 1px solid #4a314d;
  border-radius: 4px;
  padding: 0.5rem;
  font-family: 'Poppins', sans-serif;
  border-radius: 0px;
  text-align: center;
}

.card:hover,
.card:focus {
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  box-shadow: inset 0 0 0 4px #f0af4d, 0 0 1px rgba(0, 0, 0, 0);
  border: 1px solid white;
  cursor: pointer;
}

.card:active {
  background: #f0af4d;
}

.card-active {
  background: #f0af4d;
  border: 1px solid #f0af4d;
}

ul {
  list-style: none;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-inline-start: 0px;
}

h1 {
  font-size: 1.2em;
}
