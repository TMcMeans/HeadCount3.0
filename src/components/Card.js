import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Card.css';

// export const Card = ({ location, stats, handleCompareSelections }) => {
//   const data = Object.entries(stats);

//   const dataGroup = data.map(datapoint => (
//     <li className={datapoint[1] > 0.5 ? 'green' : 'red'} key={Math.random()}>
//       {datapoint[0]}: {datapoint[1]}
//     </li>
//   ));
//   return (
//     <div
//       className="card"
//       onClick={() => {
//         handleCompareSelections(location);
//       }}
//     >
//       <h1>{location}</h1>
//       <ul>{dataGroup}</ul>
//     </div>
//   );
// };

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false
    };
  }

  toggleClass = () => {
    const currState = this.state.active;
    this.setState({
      active: !currState
    });
    console.log(this.state.active);
  };

  render() {
    const { location, stats, handleCompareSelections } = this.props;
    const data = Object.entries(stats);
    const dataGroup = data.map(datapoint => (
      <li className={datapoint[1] > 0.5 ? 'green' : 'red'} key={Math.random()}>
        {datapoint[0]}: {datapoint[1]}
      </li>
    ));
    return (
      <div
        className={this.state.active ? 'card-active card' : 'card'}
        onClick={() => {
          this.toggleClass();
          handleCompareSelections(location);
        }}
      >
        <h1>{location}</h1>
        <ul>{dataGroup}</ul>
      </div>
    );
  }
}

export default Card;
