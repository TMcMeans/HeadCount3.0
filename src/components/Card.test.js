import React from 'react';
import { shallow } from 'enzyme';

import { Card } from './Card';

describe('Card', () => {
  it('should match snapshot with all html elements rendered with correct data', () => {
    const mockLocation = 'COLORADO';
    const mockStats = { 2012: 1, 2013: 1, 2014: 0, 2015: 0.552 };
    const wrapper = shallow(<Card location={mockLocation} stats={mockStats} />);

    expect(wrapper).toMatchSnapshot();
  });
});
