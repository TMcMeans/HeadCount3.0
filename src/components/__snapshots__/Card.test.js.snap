// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with all html elements rendered with correct data 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <h1>
    COLORADO
  </h1>
  <ul>
    <li
      className="green"
      key="0.5820427475701166"
    >
      2012
      : 
      1
    </li>
    <li
      className="green"
      key="0.4151093458805002"
    >
      2013
      : 
      1
    </li>
    <li
      className="red"
      key="0.9456458212112984"
    >
      2014
      : 
      0
    </li>
    <li
      className="green"
      key="0.16238363033704073"
    >
      2015
      : 
      0.552
    </li>
  </ul>
</div>
`;
